ignite/metrics/classification_report.py

```diff
@@ -0,0 +1,101 @@
+    labels: Optional[List[str]] = None,
+) -> MetricsLambda:
+    """Build a text report showing the main classification metrics. The report resembles in functionality to
+         `scikit-learn classification_report
+         <https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html#sklearn.metrics.classification_report>`_
+        The underlying implementation doesn't use the sklearn function.
+
+        Args:
+            beta: weight of precision in harmonic mean
+            output_dict: If True, return output as dict, otherwise return a str
+            output_transform: a callable that is used to transform the
+                :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+                form expected by the metric. This can be useful if, for example, you have a multi-output model and
+                you want to compute the metric with respect to one of the outputs.
+            device: optional device specification for internal storage.
+            labels: Optional list of label indices to include in the report
+            digits: Number of digits for formatting output floating point values, by default it is 5
+
+        .. code-block:: python
+
+
+            y_true = torch.tensor([[1, 0],
+            [0, 1], [1, 0], [1, 1], [1, 0], [0, 1], [0, 1], [1, 0], [0, 1], [1, 0]])
+            y_pred = torch.randint(0, 2, size=(10,))
+
+            classification_report = ClassificationReport(output_dict=True, digits=2)
+            classification_report.update((y_true, y_pred))
+            res = classification_report.compute()
+
+            # result should look like this:
+            {
+                "0": {
+                    "precision": 0.33,
+                    "recall": 0.5,
+                    "f1-score": 0.4
+                },
+                "1": {
+                    "precision": 0.5,
+                    "recall": 0.33,
+                    "f1-score": 0.4
+                },
+                "macro avg": {
+                    "precision": 0.42,
+                    "recall": 0.42,
+                    "f1-score": 0.4
+                }
+            }
+    """
+
+    # setup all the underlying metrics
+    precision = Precision(average=False, output_transform=output_transform, device=device,)
+    recall = Recall(average=False, output_transform=output_transform, device=device,)
+    fbeta = Fbeta(beta, average=False, precision=precision, recall=recall)
+    averaged_precision = precision.mean()
+    averaged_recall = recall.mean()
+    averaged_fbeta = fbeta.mean()
+
+    def _wrapper(
+        recall_metric: Metric, precision_metric: Metric, f: Metric, a_recall: Metric, a_precision: Metric, a_f: Metric,
+    ) -> Union[Collection[str], Dict]:
+        p_tensor, r_tensor, f_tensor = precision_metric, recall_metric, f
+        if p_tensor.shape != r_tensor.shape:
+            raise ValueError("Precision and Recall have mismatched shapes")
```

Comment:
In which case this can happen what user can do with that ? I think the error message is not clear about that.
If it is an internal check let's say that as 
```
f"Internal error: Precision and Recall have mismatched shapes: {p_tensor.shape} vs {r_tensor.shape}. "
"Please, open an issue with a refence on this error."
```

Reply:
Great point, updating it now.